salt/transport/tcp.py

```diff
@@ -504,6 +504,9 @@ def connect(self):
             if not self.auth.authenticated:
                 yield self.auth.authenticate()
             if self.auth.authenticated:
+                if self.auth.creds != None and self.opts['transport'] == 'tcp':
```

Comment:
I don't think we should even hit this code block _unless_ we're in the TCP transport. Did you have a reason for checking this condition explicitly?

Reply:
I did before moving this from crypt.py....Copy paste cruft.
Removed superfluous conditional from line.